### PR TITLE
improved error handling

### PR DESCRIPTION
**Pull request by bunsenstraat**
Errors did not display properly, we had issues. Mistakes with pasting addresses and so are easily made, for example when you have trailing whitespace the contract loads but then the script matching the addresses fails, and so on. Hope this brings out more clear errors to the user.

### CODE REVIEW

**Comment by FabijanC** [comment]
To prevent the trailing space issue, perhaps the address could be trimmed? It could be done before calling `this.fetchByNetwork` and `remixClient.saveFetchedToRemix` in RemixClient.ts. Or it could be done in ContractFetcher.tsx when setting `address`.

**Comment by FabijanC** [comment]
Also, what's with package-lock.json? I am aware that I updated it with my last commit, but that's what happened after installing all the dependencies before the commit. Shouldn't its content be tracked like this?

**Comment by bunsenstraat** [comment]
You can do it in every way you wish. Without the try catch block it didn't catch it. The lock file no probably not. It's errors not displaying that concern me at the moment. You want me to clean up the branch? rm the lock. About the trailing space, I'm not sure. Didn't want to meddle in that.

File: src/remix/RemixClient.ts [hunk 1]
```diff
@@ -121,7 +125,7 @@ export class RemixClient {
         try{
             response = await axios.get(`${SERVER_URL}/files/${chain}/${address}`)
         } catch(err) {
-            response = err.response;
+            throw new Error(err)
```
Comment (FabijanC) [inline src/remix/RemixClient.ts:128]: This is the block I was referring to in the earlier comment. So you claim that just the `response = await axios.get...` without an immediate catch produces an unhandled error? I mean, this way it doesn't seem wrong, just redundant.

**Review by FabijanC — CHANGES_REQUESTED** [review]
Sorry, I later realized I was not commenting in the review mode, so it might not have been clear which try-catch block I was referring to. Right now it might not be that important, but you can undo the changes to `package-lock.json`. I've done some testing and it seems to work well. Thanks for improving the error handling. If you feel like you've done enough testing, I can merge your request. Regarding the whitespace trimming, I'll probably add it in a future version together with fixes of some other old issues.

**Review by bunsenstraat — COMMENTED** [review]
should be ok now

File: src/remix/RemixClient.ts (hunk 1 above)
Comment (bunsenstraat) [reply src/remix/RemixClient.ts]: should be ok bow

**Review by FabijanC — APPROVED** [review]

**Comment by FabijanC** [comment]
@bunsenstraat Who will publish the new version to IPFS and update the `source-verification` plugin entry in the plugin repo?

**Comment by bunsenstraat** [comment]
I can do it sure.

**Pull request merged**